### PR TITLE
Support user preference to enable/disable eslint

### PR DESCRIPTION
**Pull request by smhutch**
## What kind of change does this PR introduce?

Support a new user preference. We had actually already implemented the logic for this, just missing the UI. Related to https://github.com/codesandbox/codesandbox-client/issues/3401

## What is the current behavior?

Eslint is always on

## What is the new behavior?

- Eslint can be disabled at the user level.
- The Preferences modal editor tab now uses consistent spacing and the correct heading.

## What steps did you take to test this? This is required before we can merge, make sure to test the flow you've updated.

Tested enabling / disabling the linter.

## Checklist

- [ ] Documentation
- [x] Testing 
- [x] Ready to be merged
      
- [ ] Added myself to contributors table

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/6089/commits/15e48e3889c6d71ea0a42b1596a6b2d6d5b6f98b failed.

File: packages/app/src/app/pages/common/Modals/PreferencesModal/Editor/VimModeSettings.tsx
Comment (smhutch) [inline packages/app/src/app/pages/common/Modals/PreferencesModal/Editor/VimModeSettings.tsx:41]: I changed the text here a little because users don't need to know about the need to refresh when changing this setting until after they choose to change it.

The old text was:

"Toggling the VIM extension will require a refresh. When enabled, use the command palette to control VIM"

**Review by danilowoz — APPROVED** [review]
Pretty cool! Looks good to me, but I'd wait for Ives's approval, as he has more context about how linter works here.

**Comment by OWE-AF** [comment]
@CompuIves Is there anything in particular that prevents you from approving the PR?

**Review by CompuIves — APPROVED** [review]
Oh no this just completely slipped me. Apologies, the PR looks great and can be merged!

**Pull request merged**
